.github/workflows/build.yml

```diff
@@ -0,0 +1,67 @@
+name: Build and Test
+on:
+  push:
+    branches: [main, master, develop]
+    tags: ["v*"]
+    paths:
+      - ".github/workflows/**"
+      - "packages/basemap/**"
+  pull_request:
+    branches: [main, master, develop]
+    paths:
+      - ".github/workflows/**"
+      - "packages/basemap/**"
+  workflow_dispatch: # Allows manual workflow runs
+env:
+  PKGDIR: "packages/basemap"
+  CIBW_BUILD_VERBOSITY: 1
+  PYTHONUNBUFFERED: "1"
+  PYTHONWARNINGS: "ignore:DEPRECATION"
+  GEOS_VERSION: "3.6.5"
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-2019, macos-11]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel==2.16.2
+
+      - name: Build wheels
+        env:
+          CIBW_SKIP: "pp* *-musllinux*"
+          CIBW_BEFORE_BUILD: >
+            pip install --upgrade pip
+            pip install numpy cython
+            python -c "import utils; utils.GeosLibrary('3.6.5').build('extern', njobs=16)"
```

Comment:
I forgot to mention that we actually need CMake as part of the build toolchain.

Reply:
In addition, we can remove the installation of `pip`, `numpy` and `cython` here. The `GeosLibrary` helper class is a thin wrapper to install GEOS from upstream across different platforms. The host needs a C++ toolchain and CMake.

The output of `GeosLibrary.build` is the GEOS shared library (.dll, .dylib, .so). This artifact is later used when building and repairing the `basemap` wheels.